src/client/Microsoft.Identity.Client/PlatformsCommon/Shared/AbstractPlatformProxy.cs

```diff
@@ -20,6 +20,10 @@ namespace Microsoft.Identity.Client.PlatformsCommon.Shared
 {
     internal abstract class AbstractPlatformProxy : IPlatformProxy
     {
+        public const string WindowsOS = "Windows";
+        public const string MacOs = "MacOS";
+        public const string LinuxOS = "Linux";

```

Comment:
Ah, excuse me -- we currently parse all .Net MSAL/ADAL strings with this logic:
https://msazure.visualstudio.com/One/_git/ESTS-Main?path=/src/Product/Microsoft.AzureAD.ESTS/Sts/ClientInfo.cs&version=GBmaster&line=451&lineEnd=451&lineStartColumn=17&lineEndColumn=44&lineStyle=plain&_a=contents

Again, we can change it to whatever, but we'll want to try to keep parity with existing code. E.g. it looks like we expect to be able to pull windows version out of this string with a simple regex.

Reply:
@bgavrilMS Wouldn't it be also helpful for us to know the specific Windows version in our case, like to put in logs or telemetry?

Reply:
@joelklabo - I had a look at the ESTS code and it seems you try to parse x-client-os - look for Windows + version. And if that fails, you will look at the UserAgent string (which is not reliable, particularly during refresh_token flow). I see that the os-version is only used for Windows.

I will make the change back to send the OSDescription on Windows, which seems to be "Microsoft + Windows + version". And send "MacOS" and "Linux" as x-client-os. 

I am very hesitant to make changes to `UserAgent`, because Azure SDK actually hijacks that.

Reply:
Agreed that we should not change the UserAgent. That also has impact on the UI rendering logic.
The customer issue here was entirely about the SKU OS header.